pandas/_libs/tslib.pyx

```diff
@@ -387,6 +400,10 @@ class Timestamp(_Timestamp):
                                       year or 0, month or 0, day or 0,
                                       hour), tz=hour)
 
+        if tzinfo is not None:
```

Comment:
this shouldn't be necessary if you do as the above.

Reply:
Why?  Changing line 390 above doesn't affect anything at this point in the func.

Reply:
you have the exact same check above  (if tzinfo is not none), so you shoul do this up there. (line 377)

Reply:
OK.  That requires conditioning on `elif not is_integer_object(freq)`.  Just pushed.